### PR TITLE
ADIS16497 Driver

### PR DESCRIPTION
**Pull request by mhkabir**
This brings in a driver for the ADIS16497 IMU connected over SPI.

* Sensor internally sampling at 4kHz, downsampled to 1kHz on-sensor, and read out over SPI at 1kHz.

**Test data / coverage**
Tested on a ground vehicle platform:
https://review.px4.io/plot_app?log=bd96e164-3b8b-4410-892a-abc6c3fd2af2

To be tested on aerial platforms in near future.

TODOs :
* Internal filtering is completely disabled right now. Potentially look into custom FIR filter design.
* CRC calculation table is duplicated from NuttX.
* Measurement ranges need to be set correctly for each sensor variant.

### CODE REVIEW

File: src/drivers/imu/adis16497/ADIS16497.hpp [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+#include <drivers/device/integrator.h>
+#include <lib/conversion/rotation.h>
+#include <perf/perf_counter.h>
+#include <ecl/geo/geo.h>
+
+#define ADIS16497_GYRO_DEFAULT_RATE			1000
+#define ADIS16497_GYRO_DEFAULT_DRIVER_FILTER_FREQ	80
+
+#define ADIS16497_ACCEL_DEFAULT_RATE			1000
+#define ADIS16497_ACCEL_DEFAULT_DRIVER_FILTER_FREQ	30
```
Comment (mhkabir) [inline src/drivers/imu/adis16497/ADIS16497.hpp]: Should be accessible using parameters.

File: boards/px4/fmu-v5/init/rc.board_sensors [hunk 2]
```diff
@@ -26,4 +29,6 @@ ist8310 -C -b 2 start
 ist8310 -C -b 5 start
 
 # Possible pmw3901 optical flow sensor
-pmw3901 start
+# pmw3901 start
```
Comment (mhkabir) [inline boards/px4/fmu-v5/init/rc.board_sensors]: Need to figure out way to coexist.

File: src/drivers/imu/adis16497/ADIS16497_main.cpp [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+	if (g_dev != nullptr)
+		/* if already started, the still command succeeded */
+	{
+		errx(0, "already started");
+	}
+
+	/* create the driver */
+#if defined(PX4_SPIDEV_EXTERNAL1_1)
+	g_dev = new ADIS16497(PX4_SPI_BUS_EXTERNAL1, ADIS16497_DEVICE_PATH_ACCEL, ADIS16497_DEVICE_PATH_GYRO,
+			      PX4_SPIDEV_EXTERNAL1_1, rotation);
```
Comment (mhkabir) [inline src/drivers/imu/adis16497/ADIS16497_main.cpp]: Currently using external CS1. (conflicts with default PMW3901)

File: src/modules/logger/logger.cpp [hunk 4]
```diff
@@ -725,10 +725,10 @@ void Logger::add_thermal_calibration_topics()
...
 void Logger::add_sensor_comparison_topics()
 {
-	add_topic("sensor_accel", 100);
-	add_topic("sensor_baro", 100);
-	add_topic("sensor_gyro", 100);
-	add_topic("sensor_mag", 100);
+	add_topic("sensor_accel");
+	add_topic("sensor_baro");
+	add_topic("sensor_gyro");
+	add_topic("sensor_mag");
```
Comment (mhkabir) [inline src/modules/logger/logger.cpp]: Should revert.

File: boards/px4/fmu-v5/init/rc.board_sensors (hunk 2 above)
Comment (dagar) [reply boards/px4/fmu-v5/init/rc.board_sensors]: Probably parameters to explicitly enable.

File: src/drivers/imu/adis16497/ADIS16497.hpp [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+// TODO : This is a copy of the NuttX CRC32 table
+static const uint32_t crc32_tab[] = {
+	0x00000000, 0x77073096, 0xee0e612c, 0x990951ba, 0x076dc419, 0x706af48f, 0xe963a535, 0x9e6495a3,
```
Comment (dagar) [inline src/drivers/imu/adis16497/ADIS16497.hpp]: The nuttx table didn't work?

**Comment by dagar** [comment]
I've made a few minor changes to make this safe to merge and rebased on current master.

**Pull request merged**